packages/cli/src/daemon/enable/Enable.handler.ts

```diff
@@ -207,4 +221,120 @@ export default class EnableDaemonHandler implements ICommandHandler {
                 });
         });
     }
+
+    /**
+     * Add our .zowe/bin directory to the user's PATH.
+     *
+     * @param pathToZoweBin The absolute path to our .zowe/bin drectory.
+     *
+     * @param {IAnswerQuestions} userQuestions Questions for user (if permitted)
+     *
+     * @returns {string} An informational message to display to the user after
+     *          successful completion of the operation.
+     */
+    private async addZoweBinToPath(pathToZoweBin: string, userQuestions: IAnswerQuestions): Promise<string> {
+        let userInfoMsg: string = "";
+        const osPlatform: string = ProcessUtils.getBasicSystemInfo().platform;
+
+        if (process.env.PATH?.includes(pathToZoweBin)) {
+            // bash & zsh command-path caching forces us to require a new terminal
+            if ( osPlatform !== "win32") {
+                userInfoMsg += "\n\n" + EnableDaemonHandler.openNewTerminalMsg;
+            }
+        } else {
+            // ZOWE_CLI_HOME/bin is not on our PATH, we want to add it
+            let answer: string = null;
+            if (userQuestions.addBinToPath.askUser) {
+                // alter PATH question by OS
+                let pathQuestion = "May we add the Zowe bin directory to your\nPATH in your ";
+                if ( osPlatform === "win32") {
+                    pathQuestion += "permanent user environment";
+                } else {
+                    pathQuestion += ".profile file";
+                }
+                pathQuestion += " [y or n] ? ";
+                // ask user for permission to update PATH
+                answer = await CliUtils.readPrompt(pathQuestion);
+            } else {
+                // don't ask, just use default
+                answer = userQuestions.addBinToPath.defaultVal;
+            }
+
+            if (answer !== null && answer === "y" || answer === "Y") {
+                // user wants us to do it for him/her
+                if ( osPlatform === "win32") {
+                    userInfoMsg += await this.addZoweBinOnWindows(pathToZoweBin);
+                } else {
+                    userInfoMsg += this.addZoweBinOnPosix(pathToZoweBin);
```

Comment:
Should this condition be commented out? It seems like currently if the user responds "y" to the prompt, on Linux we do nothing.

Reply:
I added the structure previously known as IAnswerQuestions, for two reasons:

1.  To enable us to more easily move the enableDaemon( ) function into an SDK if the need arises in the future. If ZoweExplorer ever needed this function, it would want to control whether or not we automatically prompt at the command line.
2. By having the EnableDaemonHandler.process( ) function call the enableDaemon( ) function with properties of `don't ask`  and `don't add-to-path`, the logic that you are pointing to does not have to be commented out, because it will never be run - we won't prompt and we will not attempt to add-to-path for Windows, Linux, or Mac (at least not yet).

I like not having to resurrect many snippets of commented code, when we decide to fully implement the add-to-path logic. With the current approach, we only have to change the structure created in the process( ) function. So, unless you have seen another flaw that I have overlooked in this technique, I would like to leave this piece of code the way that it is.

Reply:
The only operation that these functions currently perform is to correctly tell the user to open a new terminal when that is required.

Reply:
Thanks for the explanation, LGTM 🙂